### PR TITLE
fix: dependency graph for non-repl template functions

### PR DESCRIPTION
**Pull request by DanStough**
#### What type of PR is this?
kind/bug

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?
```release-note
Fixes a bug associated with Config Item dependencies not being discovered when combining Replicated and Non-Replicated template functions.
```

#### Does this PR require documentation?
NONE

### CODE REVIEW

**Review by laverya — APPROVED** [review]

File: pkg/template/depgraph_test.go [hunk 1]
```diff
@@ -201,12 +219,6 @@ func TestDepGraph(t *testing.T) {
 				"delta":   {"alpha", "charlie"},
 				"echo":    {},
 			},
-			testCAs: map[string][]string{
-				"echo": {"caA"},
-			},
-			testCAFromKeys: map[string][][2]string{
-				"delta": {{"caA", "certA"}},
-			},
```
Comment (DanStough) [inline pkg/template/depgraph_test.go:209]: This was definitely a fat-finger mistake. Will update

File: pkg/template/depgraph.go [hunk 2]
```diff
@@ -21,6 +23,17 @@ type depGraph struct {
 	CACertItemsFromKey  map[string]map[string]struct{} // items that use the TLSKeyFromCA template function. map of TLSKeyFromCA configoptions to the canames and certnames they use
 }
 
+// These functions will be used to figure out dependency order in the event standard rendering fails
+// The first argument to each is a string with the dependent config item.
+// TLS<blah> functions are deprecated and not included.
+const replFuncReExpr = `(?:ConfigOption|ConfigOptionIndex|ConfigData|ConfigOptionFilename|ConfigOptionEquals|ConfigOptionNotEquals)[[:blank:]]*\"[^\"]*\"`
```
Comment (emosbaugh) [inline pkg/template/depgraph.go:29]: how about this? change summary:
- only match space not all blanks
- one or more spaces, not zero or more
- escaping quote shouldnt be necessary
- one or more not quote, not zero or more

```suggestion
const replFuncReExpr = `(?:ConfigOption|ConfigOptionIndex|ConfigData|ConfigOptionFilename|ConfigOptionEquals|ConfigOptionNotEquals) +"[^"]+"`
```

File: pkg/template/depgraph.go [hunk 3]
```diff
@@ -21,6 +23,17 @@ type depGraph struct {
...
+
+var re *regexp.Regexp
```
Comment (emosbaugh) [inline pkg/template/depgraph.go:31]:
```suggestion
var re = regexp.MustCompile(replFuncReExpr)
```

File: pkg/template/depgraph.go [hunk 4]
```diff
@@ -21,6 +23,17 @@ type depGraph struct {
...
+
+func init() {
```
Comment (emosbaugh) [inline pkg/template/depgraph.go:33]: i think you can move this to the declaration above

**Review by emosbaugh — APPROVED** [review]
some take it or leave it comments

**Pull request merged**